azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -70,26 +77,96 @@
         }
     }
 
+    private IWebApp getWebApp(final WebAppConfig config) {
+        return az.webapp(config.getResourceGroup(), config.getAppName());
+    }
+
+    private IWebAppDeploymentSlot getDeploymentSlot(final WebAppConfig config) throws AzureExecutionException {
+        final IWebApp webApp = getWebApp(config);
+        final boolean isDeploymentSlot = StringUtils.isNotEmpty(config.getDeploymentSlotName());
+        if (isDeploymentSlot && !webApp.exists()) {
+            throw new AzureExecutionException(WEBAPP_NOT_EXIST_FOR_SLOT);
+        }
+        return webApp.deploymentSlot(config.getDeploymentSlotName());
+    }
+
+    private IWebApp createWebApp(final IWebApp webApp, final WebAppConfig webAppConfig) {
+        // todo: get or create Resource Group
+        final ResourceGroup resourceGroup = null;
+        // Get or create App Service Plan
+        final IAppServicePlan appServicePlan = az.appServicePlan(webAppConfig.getServicePlanResourceGroup(), webAppConfig.getServicePlanName());
+        if (!appServicePlan.exists()) {
+            appServicePlan.create()
+                    .withName(webAppConfig.getServicePlanName())
+                    .withResourceGroup(resourceGroup.getName())
+                    .withRegion(webAppConfig.getRegion())
+                    .withPricingTier(webAppConfig.getPricingTier())
+                    .withOperatingSystem(webAppConfig.getRuntime().getOperatingSystem())
+                    .commit();
+        }
+        return webApp.create().withName(webAppConfig.getAppName())
+                .withResourceGroup(webAppConfig.getResourceGroup())
+                .withPlan(appServicePlan.entity().getId())
+                .withRuntime(webAppConfig.getRuntime())
+                .withDockerConfiguration(webAppConfig.getDockerConfiguration())
+                .withSubscription(webAppConfig.getSubscriptionId())
+                .withAppSettings(webAppConfig.getAppSettings())
+                .commit();
+    }
+
+    private IWebApp updateWebApp(final IWebApp webApp, final WebAppConfig webAppConfig) throws AzureExecutionException {
+        // update app service plan
+        final IAppServicePlan currentPlan = webApp.plan();
+        final IAppServicePlan targetServicePlan = StringUtils.isEmpty(webAppConfig.getServicePlanName()) ? currentPlan :
+                az.appServicePlan(webAppConfig.getServicePlanResourceGroup(), webAppConfig.getServicePlanName());
+        if (!targetServicePlan.exists()) {
+            throw new AzureExecutionException(String.format("App service plan %s was not found in resource group %s",
+                    webAppConfig.getServicePlanName(), webAppConfig.getServicePlanResourceGroup()));
+        }
+        targetServicePlan.update().withPricingTier(webAppConfig.getPricingTier()).commit();
+        return webApp.update().withPlan(targetServicePlan.entity().getId())
```

Comment:
we'd better check whether it is actually changing service plan id

Reply:
Will add the full logic in later pr